hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractMessageTask.java

```diff
@@ -44,8 +44,8 @@
 
     protected final ClientMessage clientMessage;
 
-    protected final Connection connection;
-    protected ClientEndpoint endpoint;
+    protected volatile Connection connection;
```

Comment:
I did a final commit now where I update the connection during findSendConnection method, this was the original reason for it which I later forgot to update. I updated it in the latest commit now.